flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-doris/src/main/java/com/ververica/cdc/connectors/doris/sink/DorisMetadataApplier.java

```diff
@@ -0,0 +1,167 @@
+import org.apache.doris.flink.cfg.DorisOptions;
+import org.apache.doris.flink.exception.IllegalArgumentException;
+import org.apache.doris.flink.sink.schema.SchemaChangeManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/** Supports {@link DorisDataSink} to schema evolution. */
+public class DorisMetadataApplier implements MetadataApplier {
+    private static final Logger LOG = LoggerFactory.getLogger(DorisMetadataApplier.class);
+    private DorisOptions dorisOptions;
+    private SchemaChangeManager schemaChangeManager;
+
+    public DorisMetadataApplier(DorisOptions dorisOptions) {
+        this.dorisOptions = dorisOptions;
+        this.schemaChangeManager = new SchemaChangeManager(dorisOptions);
+    }
+
+    @Override
+    public void applySchemaChange(SchemaChangeEvent event) {
+        try {
+            // send schema change op to doris
+            if (event instanceof CreateTableEvent) {
+                applyCreateTableEvent((CreateTableEvent) event);
+            } else if (event instanceof AddColumnEvent) {
+                applyAddColumnEvent((AddColumnEvent) event);
+            } else if (event instanceof DropColumnEvent) {
+                applyDropColumnEvent((DropColumnEvent) event);
+            } else if (event instanceof RenameColumnEvent) {
+                applyRenameColumnEvent((RenameColumnEvent) event);
+            } else if (event instanceof AlterColumnTypeEvent) {
+                throw new RuntimeException("Unsupport schema change event, " + event);
+            }
+        } catch (Exception ex) {
+            throw new RuntimeException(
+                    "Failed to schema change, " + event + ", reason: " + ex.getMessage());
+        }
+    }
+
+    private void applyCreateTableEvent(CreateTableEvent event)
+            throws IOException, IllegalArgumentException {
+        Schema schema = event.getSchema();
+        TableId tableId = event.tableId();
+        TableSchema tableSchema = new TableSchema();
+        tableSchema.setTable(tableId.getTableName());
+        tableSchema.setDatabase(tableId.getSchemaName());
+        tableSchema.setFields(buildFields(schema));
+        tableSchema.setDistributeKeys(buildDistributeKeys(schema));
+
+        if (CollectionUtil.isNullOrEmpty(schema.primaryKeys())) {
+            tableSchema.setModel(DataModel.DUPLICATE);
+        } else {
+            tableSchema.setKeys(schema.primaryKeys());
+            tableSchema.setModel(DataModel.UNIQUE);
+        }
+        // Currently, it is not supported to pass the properties of custom tables, so temporarily
+        // hard-code some common ones first.
+        Map<String, String> props = new HashMap<>();
+        props.put("replication_num", "1");
+        props.put("light_schema_change", "true");
+        tableSchema.setProperties(props);
```

Comment:
a small suggestion, do we need to support set custom table properties by Option, which may like https://github.com/ververica/flink-cdc-connectors/pull/2765/files#diff-8abb2418ead84f1f1e2608dd03ab19ee0d0f4b74f606df3bc074d0e232efa677R66

Reply:
This is a good suggestion and has been modified.